src/pepr/operator/controllers/keycloak/authservice/authservice.ts

```diff
@@ -30,20 +30,35 @@ import {
 export const log = setupLogger(Component.OPERATOR_AUTHSERVICE);
 let lock = false;
 
-export async function authservice(pkg: UDSPackage, clients: Map<string, Client>) {
+export async function authservice(
+  pkg: UDSPackage,
+  clients: Map<string, Client>,
+): Promise<AuthserviceClient[]> {
   if (!pkg.metadata?.namespace || !pkg.metadata?.name) {
     throw new Error("Package metadata is missing required fields");
   }
 
+  // Get the requested service mesh mode, default to sidecar if not specified
+  const istioMode = pkg.spec?.network?.serviceMesh?.mode || Mode.Sidecar;
+  const previousMeshMode = pkg.status?.meshMode || Mode.Sidecar;
+
   // Get the list of clients from the package
   const authServiceClients = R.filter(
     sso => R.isNotNil(sso.enableAuthserviceSelector),
     pkg.spec?.sso || [],
   );
 
   // Check if we're in ambient mode by looking for the waypoint annotation
-  const isAmbient = pkg.spec?.network?.serviceMesh?.mode === Mode.Ambient;
+  const isAmbient = shouldUseAmbientWaypoint(pkg);
```

Comment:
I'm a bit confused by `isAmbient` here. I would think `istioMode === Mode.Ambient` would be equivalent to "is ambient", but this function also checks for any authservice clients in the Package CR. Tracing this in the code I don't think the difference is relevant/necessary - `isAmbient` is used/passed to `reconcileAuthservice` from inside of a loop over authservice clients. So we know that clients exist at that point and the istio mode should be the only relevant point.

Would also delete the comment as it's not accurate anymore I don't think.